tests/test_sanity_check/test_similar_columns.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from feature_engine.sanity_check import SimilarColumns
+
+
+def test_similar_columns_when_more_columns_in_train_than_test(df_vartypes, df_na):
+
+    # When columns are the same
+    train = df_na.copy()
+    test = df_vartypes.copy()
+
+    similar_columns = SimilarColumns()
+    similar_columns.fit(train)
+
+    transformed_df = similar_columns.transform(test)
+
+    expected_result = pd.DataFrame(
+        {
+            "Name": ["tom", "nick", "krish", "jack"],
+            "City": ["London", "Manchester", "Liverpool", "Bristol"],
+            "Studies": [np.nan, np.nan, np.nan, np.nan],
+            "Age": [20, 21, 19, 18],
+            "Marks": [0.9, 0.8, 0.7, 0.6],
+            "dob": pd.date_range("2020-02-24", periods=4, freq="T"),
+        }
+    )
+
+    pd.testing.assert_frame_equal(expected_result, transformed_df)
```

Comment:
in addition to testing the final output, we do tend to check that the attributes are learned correctly in the transfomer, for example that self.variables_ is what we expect it to be. Please have a look at the tests for other transformers in the package.